docs/glossary.md

```diff
@@ -22,13 +22,24 @@ such as the "operation store backend" for storing
 
 ## Bookmark
 
-A bookmark is a named pointer to a [commit](#commit). They automatically follow
-the commit if it gets [rewritten](#rewrite). Bookmarks are sometimes called
-"named branches" to distinguish them from [anonymous branches](#anonymous-branch).
-They are similar to Git's branches and even more similar to
+A bookmark is a named pointer to a [commit](#commit). They are similar to Git's [branches](#branch) and even more similar to
 [Mercurial's bookmarks](https://wiki.mercurial-scm.org/Bookmarks). See
 [here](bookmarks.md) for details.
 
+Unlike in Git, bookmarks automatically follow the commit if it gets
+[rewritten](#rewrite). However, bookmarks do *not* move when you create a new commit;
```

Comment:
What are some practical cases where you rewrite a commit in Git and the corresponding branch doesn’t get rewritten? I worry this is a bit misleading as‐is.

Reply:
The most important case (which I think has been addressed to a degree recently) is if you have

```
x -> branch_b -> branch_a
 \-> main
```

and you rebase `branch_a` onto `main`, `branch_b` normally does not move.

----

But I think you have a point, I'll reword this a bit to trigger people's sense of "um, actually" a bit less.

Reply:
Good example, but please mention the `--update-refs` flag (which `git rebase` learned about two years ago) so readers don't think we don't know about or that we pretend that it doesn't exist.

Reply:
I just removed "Unlike in Git" from this sentence (and moved it to another sentence).

But see also https://github.com/martinvonz/jj/pull/4654#discussion_r1802344909, which we could also fix.

Reply:
I think it makes sense to leave it for another commit/PR.